Settlement/index.ts

```diff
@@ -49,7 +52,17 @@ export namespace Settlement {
 					result.entries.failed ? result.entries.failed.count++ : (result.entries.failed = { count: 1 })
 					break
 			}
+			if (entry.type == "unknown") {
+				;(result.warnings ??= []).push(Warning.UnknownEntry.create(entry, settlement.id))
+			}
 		}
+		Object.entries(result.totals).forEach(
```

Comment:
The compile function is called many times. I think we should only check if the settlement is negative at the end, somewhere around payout.